xla/service/gpu/transforms/collective_permute_cycle_decomposer_test.cc

```diff
@@ -187,6 +206,76 @@ TEST_F(CollectivePermuteCycleDecomposerTest, ForwardCycleWithMatmul) {
               HasSubstr("_xla_send_recv_validation={{0,7},{1,8},{2,9}}"));
 }
 
+TEST_F(CollectivePermuteCycleDecomposerTest, ForwardCycleWithMatmulNoChannel) {
```

Comment:
Does this transform depend on the surrounding while loop? If not, I don't think we need to add test cases with while loops